File: translator-projects-poetry/translator-node/src/test_data.js
import fetch from "node-fetch";
import dotenv from "dotenv";
dotenv.config();
const BASE_URL = process.env.BASE_URL || 'http://localhost:8000';
console.log('Using BASE_URL =', BASE_URL);

const data = [
  { word: 'sol', translation: 'sun' },
  { word: 'luna', translation: 'moon' },
  { word: 'estrella', translation: 'star' }
];

(async () => {
  console.log('Adding words...');
  const addRes = await fetch(`${BASE_URL}/add_words`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data)
  });
  console.log('Add:', await addRes.json());

  const single = await fetch(`${BASE_URL}/translate?word=sol`);
  console.log('Single:', await single.json());

  const many = await fetch(`${BASE_URL}/translate_many`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(['sol','luna','estrella'])
  });
  console.log('Many:', await many.json());
})();

File: translator-projects-poetry/translator-python/app/test_data.py
import os, requests, time
BASE_URL = os.getenv("BASE_URL", "http://localhost:8000")
print("Using BASE_URL =", BASE_URL)

data = [
    {"word": "gato", "translation": "cat"},
    {"word": "árbol", "translation": "tree"},
    {"word": "cielo", "translation": "sky"}
]
print("Adding words...")
r = requests.post(f"{BASE_URL}/add_words", json=data)
print("Add response:", r.status_code, r.json())

time.sleep(0.5)
print("Translate single 'gato'")
r = requests.get(f"{BASE_URL}/translate", params={"word": "gato"})
print(r.json())

print("Translate many ['gato','perro','cielo']")
r = requests.post(f"{BASE_URL}/translate_many", json=['gato','perro','cielo'])
print(r.json())
